components/script/dom/htmlelement.rs

```diff
@@ -302,6 +335,20 @@ impl HTMLElement {
         let local_name = Atom::from_slice(&to_snake_case(local_name));
         element.remove_attribute(&ns!(""), &local_name);
     }
+
+    pub fn supported_prop_names_custom_attr(&self) -> Vec<DOMString> {
+        let element = ElementCast::from_ref(self);
+        element.attrs().iter().map(JS::root).map(|attr| {
+            let raw_name = (**attr.r().local_name()).to_owned();
+            to_camel_case(raw_name)
+        }).fold(Vec::new(), |mut acc, opt| {
```

Comment:
I think you may be able to use https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map and collect the results